terraso_backend/apps/soil_id/graphql/soil_id/resolvers.py

```diff
@@ -0,0 +1,272 @@
+    )
+
+
+def resolve_soil_match_info(score: float, rank: str):
+    return SoilMatchInfo(score=score, rank=int(rank) - 1)
+
+
+def resolve_location_based_soil_match(soil_match: dict):
+    soil_id = soil_match["id"]
+    site_data = soil_match["site"]["siteData"]
+
+    return LocationBasedSoilMatch(
+        data_source=site_data["dataSource"],
+        distance_to_nearest_map_unit_m=site_data["minCompDistance"],
+        match=resolve_soil_match_info(soil_id["score_loc"], soil_id["rank_loc"]),
+        soil_info=resolve_soil_info(soil_match),
+    )
+
+
+def resolve_location_matches_from_soil_id_result(soil_list_json: dict):
+    matches = []
+    for match in soil_list_json["soilList"]:
+        if match["id"]["rank_loc"] != "Not Displayed":
+            matches.append(resolve_location_based_soil_match(match))
+    return LocationBasedSoilMatches(matches=matches)
+
+
+def resolve_location_based_soil_matches(_parent, _info, latitude: float, longitude: float):
+    # TODO: remove this line to re-enable using the actual algorithm to resolve this query
+    return dummy_location_matches
+
+    result = list_soils(lat=latitude, lon=longitude)
+
+    if isinstance(result, str):
+        return None
+
+    return resolve_location_matches_from_soil_id_result(result.soil_list_json)
+
+
+def resolve_data_based_soil_match(soil_matches: list[dict], ranked_match: dict):
+    soil_match = [
+        match
+        for match in soil_matches
+        if int(match["site"]["siteData"]["componentID"]) == ranked_match["componentID"]
+    ][0]
+    site_data = soil_match["site"]["siteData"]
+
+    return DataBasedSoilMatch(
+        data_source=site_data["dataSource"],
+        distance_to_nearest_map_unit_m=site_data["minCompDistance"],
+        location_match=resolve_soil_match_info(ranked_match["score_loc"], ranked_match["rank_loc"]),
+        data_match=resolve_soil_match_info(ranked_match["score_data"], ranked_match["rank_data"]),
+        combined_match=resolve_soil_match_info(
+            ranked_match["score_data_loc"], ranked_match["rank_data_loc"]
+        ),
+        soil_info=resolve_soil_info(soil_match),
+    )
+
+
+def parse_texture(texture: Optional[DepthDependentSoilData.Texture]):
+    if texture is None:
+        return None
+    return texture.value.replace("_", " ").lower()
+
+
+def parse_rock_fragment_volume(
+    rock_fragment_volume: Optional[DepthDependentSoilData.RockFragmentVolume],
+):
+    if rock_fragment_volume is None:
+        return None
+    elif rock_fragment_volume == DepthDependentSoilData.RockFragmentVolume.VOLUME_0_1:
+        return "0-1%"
+    elif rock_fragment_volume == DepthDependentSoilData.RockFragmentVolume.VOLUME_1_15:
+        return "1-15%"
+    elif rock_fragment_volume == DepthDependentSoilData.RockFragmentVolume.VOLUME_15_35:
+        return "15-35%"
+    elif rock_fragment_volume == DepthDependentSoilData.RockFragmentVolume.VOLUME_35_60:
+        return "35-60%"
+    else:
+        return ">60%"
```

Comment:
suggestion: use constants for the return values.

Reply:
see: https://github.com/techmatters/terraso-backend/pull/1329#discussion_r1646771217